src/modules/profile/sagas.ts

```diff
@@ -134,15 +137,18 @@ export function createProfileSaga({
     const profileMetadata: Profile = {
       avatars: [newAvatar, ...profileWithContentHashes.avatars.slice(1)]
     }
-
-    yield call(
-      [entities, 'deployEntityWithoutNewFiles'],
-      profileMetadata,
-      getHashesByKeyMap(newAvatar),
-      EntityType.PROFILE,
-      address,
-      address
-    )
+    const identity = getIdentity()
```

Comment:
With the new code:
- If there's an identity, the `deployEntityWithoutNewFiles` is called, meaning that the entity is deployed.
- If there's no identity, the `deployEntityWithoutNewFiles` is not called and nothing happens.

In the older code, if the deployment failed, the  the update process failed as well, but this is not the case anymore. We should throw if there's no identity set, what do you think?